Tests/DependencyInjection/Compiler/AnnotationCompilerPassTest.php

```diff
@@ -90,6 +90,21 @@ public function testMinimalAdmin()
         );
     }
 
+    public function testIdForAdmin()
+    {
+        /*
+         * @Admin(class="Sonata\AdminBundle\Entity\Foo", id="my.id")
+         */
+        $annotation = new Admin();
+        $annotation->class = 'Sonata\AdminBundle\Entity\Foo';
+
+        $meta = new ClassMetadata('Sonata\AdminBundle\Tests\Fixtures\Entity\Foo');
+
+        $annotation->processMetadata($meta);
+
+        $this->assertSame($meta->id == 'my.id');
```

Comment:
- although it does not matter here, we usually prefer `assertSame` in Sonata (maybe not from the beginning, but now we do)
- the expected value should be used as first argument if you want good error messages in case of failure

so `$this->assertSame('my.id', $meta->id)`